speech/google/cloud/speech/client.py

```diff
@@ -188,79 +228,141 @@ def sync_recognize(self, sample, language_code=None,
         else:
-            raise ValueError('result in api should have length 1')
-
-
-def _build_request_data(sample, language_code=None, max_alternatives=None,
-                        profanity_filter=None, speech_context=None):
-    """Builds the request data before making API request.
-
-    :type sample: :class:`~google.cloud.speech.sample.Sample`
-    :param sample: Instance of ``Sample`` containing audio information.
-
-    :type language_code: str
-    :param language_code: (Optional) The language of the supplied audio as
-                          BCP-47 language tag. Example: ``'en-GB'``.
-                          If omitted, defaults to ``'en-US'``.
-
-    :type max_alternatives: int
-    :param max_alternatives: (Optional) Maximum number of recognition
-                             hypotheses to be returned. The server may
-                             return fewer than maxAlternatives.
-                             Valid values are 0-30. A value of 0 or 1
-                             will return a maximum of 1. Defaults to 1
-
-    :type profanity_filter: bool
-    :param profanity_filter: If True, the server will attempt to filter
-                             out profanities, replacing all but the
-                             initial character in each filtered word with
-                             asterisks, e.g. ``'f***'``. If False or
-                             omitted, profanities won't be filtered out.
-
-    :type speech_context: list
-    :param speech_context: A list of strings (max 50) containing words and
-                           phrases "hints" so that the speech recognition
-                           is more likely to recognize them. This can be
-                           used to improve the accuracy for specific words
-                           and phrases. This can also be used to add new
-                           words to the vocabulary of the recognizer.
-
-    :rtype: dict
-    :returns: Dictionary with required data for Google Speech API.
-    """
-    if sample.content is not None:
-        audio = {'content':
-                 _bytes_to_unicode(b64encode(_to_bytes(sample.content)))}
-    else:
-        audio = {'uri': sample.source_uri}
-
-    config = {'encoding': sample.encoding,
-              'sampleRate': sample.sample_rate}
-
-    if language_code is not None:
-        config['languageCode'] = language_code
-    if max_alternatives is not None:
-        config['maxAlternatives'] = max_alternatives
-    if profanity_filter is not None:
-        config['profanityFilter'] = profanity_filter
-    if speech_context is not None:
-        config['speechContext'] = {'phrases': speech_context}
-
-    data = {
-        'audio': audio,
-        'config': config,
-    }
-
-    return data
+            raise IndexError('More than one result returned from API.')
+
+    @staticmethod
+    def _build_request_data(sample, language_code=None, max_alternatives=None,
```

Comment:
This was fine as a function in the module. Making it a static method doesn't do anything useful here and above makes the reference thornier: `_JSONSpeechAPI._build_request_data` instead of `_build_request_data`